pandas/core/generic.py

```diff
@@ -721,12 +721,37 @@ def transpose(self, *args, **kwargs):
         new_axes = self._construct_axes_dict_from(
             self, [self._get_axis(x) for x in axes_names]
         )
-        new_values = self.values.transpose(axes_numbers)
-        if kwargs.pop("copy", None) or (len(args) and args[-1]):
-            new_values = new_values.copy()
+
+        if (
+            self._is_homogeneous_type
+            and len(self._data.blocks)
+            and is_extension_array_dtype(self._data.blocks[0].dtype)
+        ):
+            kwargs.pop("copy", None)  # by definition, we're copying
+            dtype = self._data.blocks[0].dtype
+            arr_type = dtype.construct_array_type()
+
```

Comment:
I would move this logic to pandas/core/reshape/reshape.py this has a lot of similiarity to _unstack_extension_series